[Bug]: [Nightly]Search binary superstructure with index type 'BIN_FLAT' return result missing data

## Environment
- Milvus version:04e595d
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    kafka
- SDK version(e.g. pymilvus v2.0.0rc2):2.2.1.dev5
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others: 

## Current Behavior:
Search binary superstructure with index type 'BIN_FLAT' return result missing data.
```
[2023-01-11T21:26:49.219Z] [2023-01-11 20:28:26 - ERROR - ci_test]: search_results_check: limit(topK) searched (8) is not equal with expected (10) (func_check.py:227)
```

## Expected Behavior:
succeed

## Report 
https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/2.2.0/61/pipeline/140/

1. log: [artifacts-milvus-distributed-kafka-nightly-61-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/2.2.0/61/artifact/artifacts-milvus-distributed-kafka-nightly-61-pymilvus-e2e-logs.tar.gz)
2. collection name: search_collection_01wDj7kl
3. failed time: [2023-01-11T20:28:27.610Z] [gw0] [ 53%] FAILED testcases/test_search.py::TestCollectionSearch::test_search_binary_superstructure_flat_index[True-False-True-BIN_FLAT]
4. case details:
```
[2023-01-11T21:26:49.217Z] _ TestCollectionSearch.test_search_binary_superstructure_flat_index[True-False-True-BIN_FLAT] _
[2023-01-11T21:26:49.217Z] [gw0] linux -- Python 3.8.16 /usr/local/bin/python3
[2023-01-11T21:26:49.217Z] 
[2023-01-11T21:26:49.217Z] self = <test_search.TestCollectionSearch object at 0x7fb2a74817c0>
[2023-01-11T21:26:49.217Z] auto_id = True, _async = False, index = 'BIN_FLAT', is_flush = True
[2023-01-11T21:26:49.217Z] 
[2023-01-11T21:26:49.217Z]     @pytest.mark.tags(CaseLabel.L2)
[2023-01-11T21:26:49.217Z]     @pytest.mark.parametrize("index", ["BIN_FLAT"])
[2023-01-11T21:26:49.217Z]     def test_search_binary_superstructure_flat_index(self, auto_id, _async, index, is_flush):
[2023-01-11T21:26:49.217Z]         """
[2023-01-11T21:26:49.217Z]         target: search binary_collection, and check the result: distance
[2023-01-11T21:26:49.217Z]         method: compare the return distance value with value computed with SUPERSTRUCTURE
[2023-01-11T21:26:49.217Z]                 (1) The returned limit(topK) are impacted by dimension (dim) of data
[2023-01-11T21:26:49.217Z]                 (2) Searched topK is smaller than set limit when dim is large
[2023-01-11T21:26:49.217Z]                 (3) It does not support "BIN_IVF_FLAT" index
[2023-01-11T21:26:49.217Z]                 (4) Only two values for distance: 0 and 1, 0 means hits, 1 means not
[2023-01-11T21:26:49.217Z]         expected: the return distance equals to the computed value
[2023-01-11T21:26:49.217Z]         """
[2023-01-11T21:26:49.217Z]         # 1. initialize with binary data
[2023-01-11T21:26:49.217Z]         nq = 1
[2023-01-11T21:26:49.217Z]         dim = 8
[2023-01-11T21:26:49.217Z]         collection_w, _, binary_raw_vector, insert_ids, time_stamp \
[2023-01-11T21:26:49.217Z]             = self.init_collection_general(prefix, True, default_nb, is_binary=True, auto_id=auto_id,
[2023-01-11T21:26:49.217Z]                                            dim=dim, is_index=True, is_flush=is_flush)[0:5]
[2023-01-11T21:26:49.217Z]         # 2. create index
[2023-01-11T21:26:49.217Z]         default_index = {"index_type": index, "params": {"nlist": 128}, "metric_type": "SUPERSTRUCTURE"}
[2023-01-11T21:26:49.217Z]         collection_w.create_index("binary_vector", default_index)
[2023-01-11T21:26:49.217Z]         collection_w.load()
[2023-01-11T21:26:49.217Z]         # 3. generate search vectors
[2023-01-11T21:26:49.217Z]         _, binary_vectors = cf.gen_binary_vectors(nq, dim)
[2023-01-11T21:26:49.217Z]         # 4. search and compare the distance
[2023-01-11T21:26:49.217Z]         search_params = {"metric_type": "SUPERSTRUCTURE", "params": {"nprobe": 10}}
[2023-01-11T21:26:49.217Z] >       res = collection_w.search(binary_vectors[:nq], "binary_vector",
[2023-01-11T21:26:49.217Z]                                   search_params, default_limit, "int64 >= 0",
[2023-01-11T21:26:49.217Z]                                   _async=_async,
[2023-01-11T21:26:49.217Z]                                   travel_timestamp=0,
[2023-01-11T21:26:49.217Z]                                   check_task=CheckTasks.check_search_results,
[2023-01-11T21:26:49.217Z]                                   check_items={"nq": nq,
[2023-01-11T21:26:49.217Z]                                                "ids": insert_ids,
[2023-01-11T21:26:49.217Z]                                                "limit": default_limit,
[2023-01-11T21:26:49.217Z]                                                "_async": _async})[0]
[2023-01-11T21:26:49.217Z] 
[2023-01-11T21:26:49.217Z] testcases/test_search.py:1998: 
```

--- a/tests/python_client/testcases/test_collection.py
+++ b/tests/python_client/testcases/test_collection.py
@@ -986,7 +986,6 @@ class TestCollectionParams(TestcaseBase):
                                              check_items={exp_name: c_name, exp_schema: schema})
 
     @pytest.mark.tags(CaseLabel.L2)
-    @pytest.mark.xfail(reason="issue #21238")
     def test_create_collection_over_maximum_fields(self):
         """
         target: Test create collection with more than the maximum fields
@@ -1690,7 +1689,6 @@ class TestCreateCollectionInvalid(TestcaseBase):
     """
 
     @pytest.mark.tags(CaseLabel.L2)
-    @pytest.mark.xfail(reason="issue #21238")
     def test_create_collection_limit_fields(self):
         """
         target: test create collection with maximum fields
@@ -2243,7 +2241,7 @@ class TestLoadCollection(TestcaseBase):
         yield request.param
 
     @pytest.mark.tags(CaseLabel.L2)
-    @pytest.mark.xfail("issue #21618")
+    @pytest.mark.xfail(reason="issue #21618")
     def test_load_replica_non_number(self, get_non_number_replicas):
         """
         target: test load collection with non-number replicas

--- a/tests/python_client/testcases/test_partition.py
+++ b/tests/python_client/testcases/test_partition.py
@@ -316,7 +316,7 @@ class TestPartitionParams(TestcaseBase):
         yield request.param
 
     @pytest.mark.tags(CaseLabel.L2)
-    @pytest.mark.xfail("issue #21618")
+    @pytest.mark.xfail(reason="issue #21618")
     def test_load_partition_replica_non_number(self, get_non_number_replicas):
         """
         target: test load partition with non-number replicas

--- a/tests/python_client/testcases/test_search.py
+++ b/tests/python_client/testcases/test_search.py
@@ -2021,15 +2021,11 @@ class TestCollectionSearch(TestcaseBase):
         res = collection_w.search(binary_vectors[:nq], "binary_vector",
                                   search_params, default_limit, "int64 >= 0",
                                   _async=_async,
-                                  travel_timestamp=0,
-                                  check_task=CheckTasks.check_search_results,
-                                  check_items={"nq": nq,
-                                               "ids": insert_ids,
-                                               "limit": default_limit,
-                                               "_async": _async})[0]
+                                  travel_timestamp=0)[0]
         if _async:
             res.done()
             res = res.result()
+        assert len(res[0]) <= default_limit
         assert res[0].distances[0] == 0.0
 
     @pytest.mark.tags(CaseLabel.L2)

